egs/librispeech/ASR/pruned_transducer_stateless2/conformer.py

```diff
@@ -158,7 +158,7 @@ def forward(
         if not is_jit_tracing():
             assert x.size(0) == lengths.max().item()
 
-        src_key_padding_mask = make_pad_mask(lengths)
+        src_key_padding_mask = make_pad_mask(lengths, x.size(0))
```

Comment:
Does `x.size(0)` equals to `lengths.max()`?

Reply:
When we do inference with fixed-length and batched inputs, the x.size(0) might not match the lengths.max()
You can see how we send audios in our client: https://github.com/yuekaizhang/Triton-ASR-Client/blob/main/client.py#L340
If we pad the audios with large size of lengths.max(), it will give errors.
So we create the PR to fix the issue in this scenario.

Reply:
Line 159 ensures that x.size(0) == lengths.max().

When you pad the samples in triton server, could you also update `lengths` accordingly?

Reply:
The lengths should always be the actual length of input audios. We pad the audio to the nearest 10 seconds then send it to the server. This will not affect training. 
For example, we have audios lengths = [9.5, 6.0, 7.6] seconds and we will pad it to 10 seconds. The x.size(0)=10, but the lengths.max()=9.5. This will give errors when we do inference.

Reply:
I see. Thanks for the explanation.

Reply:
Yeah,  we have cases that x.size(0) > lengths.max().item() during triton inference, since audios from different users has to be padded to the same length in order to do batch inference. So we have to pad them at client side.